internal/promote/storage.go

```diff
@@ -98,37 +98,72 @@ func (prs PackageVersions) Strings() []string {
 
 // CloneRepository method clones the repository and changes branch to stage.
 func CloneRepository(user, stage string) (*git.Repository, error) {
-	r, err := git.Clone(memory.NewStorage(), memfs.New(), &git.CloneOptions{
-		URL:           fmt.Sprintf(repositoryURL, "elastic"),
-		RemoteName:    remoteName,
-		ReferenceName: plumbing.NewBranchReferenceName(stage),
+	// Initialize repository
+	r, err := git.Init(memory.NewStorage(), memfs.New())
+	if err != nil {
+		return nil, errors.Wrap(err, "initializing repository")
+	}
+
+	// Add remotes
+	userRepositoryURL := fmt.Sprintf(repositoryURL, user)
+	userRemote, err := r.CreateRemote(&config.RemoteConfig{
+		Name: user,
+		URLs: []string{
+			userRepositoryURL,
+		},
 	})
 	if err != nil {
-		return nil, errors.Wrap(err, "cloning package-storage repository failed")
+		return nil, errors.Wrap(err, "creating user remote failed")
+	}
+	remote, err := r.CreateRemote(&config.RemoteConfig{
```

Comment:
Maybe call this `upstreamRemote` so it parallel's `userRemote`? In the same vein maybe we should rename `remoteName` to `upstream`?

Reply:
Renamed.